### PR TITLE
Add 'redundant enumerate()' violation.

### LINKED ISSUE #1825: Detect redundant enumerate
# Rule request

## Thesis

```python
# bad
for _, item in enumerate(items):
  ...

# good
for item in items:
  ...
```

## Reasoning

Simplification.

P.S. I'm not making it, have seen it in the wild 🙃

### PR DESCRIPTION
**Pull request by lensvol**
Closes #1825

# I have made things!

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues

🙏 Please, if you or your company is finding wemake-python-styleguide valuable, help us sustain the project by sponsoring it transparently on https://opencollective.com/wemake-python-styleguide. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: wemake_python_styleguide/violations/best_practices.py [hunk 1]
```diff
@@ -2608,3 +2610,33 @@ def smth():
...
+
+        # Correct:
+        for item in items:
+          ...
+
+        # Wrong:
+        for _, item in enumerate(items):
+          ...
+
+    .. versionadded:: 0.18.0
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/best_practices.py:2637]: 0.16.0

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot!

File: tests/test_checker/test_noqa.py [hunk 2]
```diff
@@ -247,6 +247,7 @@
     'WPS465': 1,
     'WPS466': 0,  # defined in version specific table.
     'WPS467': 1,
+    'WPS468': 1,
```
[the remark this answers is not in the document]
Comment (lensvol) [reply tests/test_checker/test_noqa.py:250]: Should I fix that as well?

**Comment by sobolevn** [comment]
Thanks a lot!

**Pull request merged**